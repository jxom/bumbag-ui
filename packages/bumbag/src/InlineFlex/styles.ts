import { cssClass } from '../styled';
import { theme } from '../utils';

export const InlineFlex = (styleProps) => cssClass`
  display: inline-flex;

  & {
    ${theme(styleProps.themeKey, `styles.base`)(styleProps)};
  }
`;
