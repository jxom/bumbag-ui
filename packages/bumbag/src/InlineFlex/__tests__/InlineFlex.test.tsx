import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { InlineFlex } from '../InlineFlex';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(<InlineFlex>Hello world</InlineFlex>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(
      <InlineFlex position="absolute" top="0">
        Hello world
      </InlineFlex>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('composition', () => {
  describe('as', () => {
    it('should render correctly', () => {
      const { container } = render(<InlineFlex use="p">Hello world</InlineFlex>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('hook', () => {
    it('should return with InlineFlex props', () => {
      const { result } = renderHook(() => InlineFlex.useProps());
      expect(result.current).toMatchSnapshot();
    });
  });

  describe('render props', () => {
    it('should render correctly', () => {
      const { container } = render(
        <InlineFlex>{(InlineFlexProps) => <div {...InlineFlexProps}>Hello world</div>}</InlineFlex>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('theming', () => {
  it('InlineFlex.base should render correctly', () => {
    const { container } = render(<InlineFlex>hello world</InlineFlex>, {
      theme: { InlineFlex: { styles: { base: { backgroundColor: 'red' } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for className', () => {
    const { container } = render(<InlineFlex>hello world</InlineFlex>, {
      theme: { InlineFlex: { defaultProps: { className: 'test' } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
