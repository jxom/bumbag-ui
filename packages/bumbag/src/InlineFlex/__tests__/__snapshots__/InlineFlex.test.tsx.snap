// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`composition as should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

<p
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-InlineFlex"
>
  Hello world
</p>
`;

exports[`composition hook should return with InlineFlex props 1`] = `
Object {
  "className": "css-269jkv css-63it0p css-jwhr8m fp-Box fp-InlineFlex",
}
`;

exports[`composition render props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-InlineFlex"
>
  Hello world
</div>
`;

exports[`defaultProps should render correctly for className 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

<div
  class="emotion-0 emotion-1 emotion-2 test fp-Box fp-InlineFlex"
>
  hello world
</div>
`;

exports[`props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-InlineFlex"
>
  Hello world
</div>
`;

exports[`props should render correctly with CSS props 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-2 {
  position: absolute !important;
  top: 0 !important;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-InlineFlex"
>
  Hello world
</div>
`;

exports[`theming InlineFlex.base should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-0 {
  background-color: #ff0000;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-InlineFlex"
>
  hello world
</div>
`;
